src/plugins/implicitRelationships.ts

```diff
@@ -0,0 +1,122 @@
+*/
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { findByProps, findStoreLazy } from "@webpack";
+import { ChannelStore, FluxDispatcher, GuildStore, RelationshipStore, UserStore } from "@webpack/common";
+
+const UserAffinitiesStore = findStoreLazy("UserAffinitiesStore");
+
+export default definePlugin({
+    name: "ImplicitRelationships",
+    description: "Shows your implicit relationships in the Friends tab.",
+    authors: [Devs.Dolfies],
+    patches: [
+        {
+            find: "FriendsEmptyState: Invalid empty state",
+            replacement: [
+                // Counts header
+                {
+                    match: /toString\(\)\}\);case ([^;]+)\.([^;]+)\.BLOCKED/,
+                    replace: "toString()});case $1.$2.IMPLICIT:return \"Implicit — \"+t.toString();case $1.$2.BLOCKED"
+                },
+                // No friends page
+                {
+                    match: /case ([^;]+)\.([^;]+)\.ONLINE:return ([^;]+)\.SECTION_ONLINE/,
+                    replace: "case $1.$2.ONLINE:case $1.$2.IMPLICIT:return $3.SECTION_ONLINE"
+                },
+                // Sections header
+                {
+                    match: /\(0,(.{1})\.jsx\)\((.{1})\.TabBar\.Item,\{id:(.{1})\.(.{1,3})\.BLOCKED,className:([^\s]+)\.item,children:([^;]+)\.([^;]+)\.Messages\.BLOCKED\}\)/,
+                    replace: "(0,$1.jsx)($2.TabBar.Item,{id:$3.$4.IMPLICIT,className:$5.item,children:\"Implicit\"}),$&"
+                },
+                // Sections content
+                {
+                    match: /case ([^;]+)\.([^;]+)\.BLOCKED:return ([^;]+)\.type===([^;]+)\.([^;]+)\.BLOCKED/,
+                    replace: "case $1.$2.BLOCKED:return $3.type===$4.$5.BLOCKED;case $1.$2.IMPLICIT:return $3.type===5"
+                },
+                // Piggyback relationship fetch
+                {
+                    match: /([^;]+)\.([^;]+)\.fetchRelationships\(\)/,
+                    // This relationship fetch is actually completely useless, but whatevs
+                    replace: "$1.$2.fetchRelationships(),$self.fetchImplicitRelationships()"
+                },
+            ],
+        },
+    ],
+
+    hasDM(userId: string): boolean {
+        const privateChannels = ChannelStore.getSortedPrivateChannels();
+        return privateChannels.some(channel => channel.recipients.includes(userId));
+    },
+
+    async fetchImplicitRelationships() {
+        // Implicit relationships are defined as users that you:
+        // 1. Have an affinity for
+        // 2. Do not have a relationship with // TODO: Check how this works with pending/blocked relationships
+        // 3. Have a mutual guild with
+        const userAffinities: Set<string> = UserAffinitiesStore.getUserAffinitiesUserIds();
+        const nonFriendAffinities = Array.from(userAffinities).filter(
+            id => !RelationshipStore.getRelationshipType(id)
```

Comment:
shouldn't this be
```suggestion
            id => RelationshipStore.getRelationshipType(id) === 5 /* IMPLICIT */
```
![image](https://github.com/Vendicated/Vencord/assets/45497981/1565d147-9575-4bc1-a6d8-b0b87e77abe6)

Reply:
No, the whole point of this function is to figure out what users Discord treats as implicit relationships and insert them into cache. `nonFriendAffinities` is just user affinities that you do not have an existing *real* relationship with.

Implicit relationships aren't actually stored as relationships on Discord's end, so they aren't sent in READY or normal `GET /users/@me/relationships` calls. Implicit relationships themselves are a GameSDK artifact that was kept alive when they added the `relationships.read` OAuth2 scope (if you `GET /users/@me/relationships?with_implicit=true` with a bearer token with said scope, Discord calculates your implicit relationships and returns them as type 5).

The client backfills type 5 relationships when it detects them from PRESENCE_UPDATE events, and also caches all of them when the GameSDK's GET_RELATIONSHIPS is called (though it has the logic slightly wrong 😭).